docs/notebooks/asynchronous_batch_bo.pct.py

```diff
@@ -0,0 +1,197 @@
+#
+# Together these two notebooks give a comprehensive overview of how to use Trieste in asynchronous scenarios.
+
+# %%
+# silence TF warnings and info messages, only print errors
+# https://stackoverflow.com/questions/35911252/disable-tensorflow-debugging-information
+import os
+os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"
+import tensorflow as tf
+tf.get_logger().setLevel("ERROR")
+
+import ray
+import numpy as np
+import time
+
+# %% [markdown]
+# Just as before, we use Branin function with delays.
+
+# %%
+from trieste.objectives import scaled_branin
+
+def objective(points, sleep=True):
+    if points.shape[1] != 2:
+        raise ValueError(f"Incorrect input shape, expected (*, 2), got {points.shape}")
+
+    observations = []
+    for point in points:
+        observation = scaled_branin(point).numpy()
+        if sleep:
+            # insert some artificial delay
+            # increases linearly with the absolute value of points
+            # which means our evaluations will take different time
+            delay = 3 * np.sum(point)
+            time.sleep(delay)
+        observations.append((point, observation))
+
+    return observations
+
+
+# %%
+objective(np.array([[0.1, 0.5]]), sleep=False)
+
+
+# %% [markdown]
+# To turn our objective function into a Ray task, we wrap it in a function with appropriate decorator. We are not using anything beyond Ray tasks API in this tutorial, and refer interested readers to [Ray documentation](https://docs.ray.io/en/latest/walkthrough.html) and [Ray crash course](https://github.com/anyscale/academy/blob/main/ray-crash-course/01-Ray-Tasks.ipynb) for more details.
+
+# %%
+@ray.remote
+def ray_objective(points, sleep=True):
+    return objective(points, sleep)
+
+
+# %% [markdown]
+# We prepare the model and some initial data to kick-start the optimization process.
+
+# %%
+from trieste.space import Box
+from trieste.data import Dataset
+
+search_space = Box([0, 0], [1, 1])
+num_initial_points = 3
+initial_query_points = search_space.sample(num_initial_points)
+initial_observations = objective(initial_query_points, sleep=False)
+initial_data = Dataset(
+    query_points=initial_query_points,
+    observations=tf.constant([x[1] for x in initial_observations], dtype=tf.float64),
+)
+
+import gpflow
+from trieste.models.gpflow import GaussianProcessRegression
+
+
+def build_model(data):
+    variance = tf.math.reduce_variance(data.observations)
+    kernel = gpflow.kernels.RBF(variance=variance)
+    gpr = gpflow.models.GPR(data.astuple(), kernel, noise_variance=1e-5)
+    gpflow.set_trainable(gpr.likelihood, False)
+    return GaussianProcessRegression(gpr)
+
+
+# %% [markdown]
+# Here we set some parameters of the optimization run. See comments below for explanation of what each one means.
```

Comment:
Here we set **up the configuration** of **our** optimization run. See comments below for **details**